flatflow/data/dataset_test.cc

```diff
@@ -92,4 +99,121 @@ TEST(DatasetTest, Constructor) {
   EXPECT_TRUE(dataset.empty());
 }
 
+TEST(DatasetTest, IntraShuffle) {
+  // Test case for IntraShuffle
+  //
+  // NOTE:
+  //
+  // * We first check if the dataset vectors are sorted.
+  // * After shuffling, we check if dataset vectors are not sorted.
+  // * Then using the same input vector, we shuffle the slots.
+  // * Last, we check if the slots and dataset vectors are equal. 
+
+  std::srand(static_cast<unsigned int>(std::time(nullptr)));
+  thread_local auto generator = std::ranlux48();
```

Comment:
``generator`` is used in line 202, so it seems better to initialize it right before use.